redisvl/index.py

```diff
@@ -5,11 +5,11 @@
 if TYPE_CHECKING:
     from redis.commands.search.field import Field
     from redis.commands.search.result import Result
+    from redisvl.query.query import BaseQuery
```

Comment:
For my own understanding, can you help clarify when types should be listed under TYPE_CHECKING and then "<type>" format is used?

Reply:
When they are only used to decorate messages for mypy